packages/eddsa-poseidon/src/eddsa-poseidon.ts

```diff
@@ -87,7 +87,7 @@ export function derivePublicKey(privateKey: Buffer | Uint8Array | string): Point
  * @param message The message to be signed.
  * @returns The signature object, containing properties relevant to EdDSA signatures, such as 'R8' and 'S' values.
  */
-export function signMessage(privateKey: Buffer | Uint8Array | string, message: BigNumberish): Signature<bigint> {
+export function signMessage(privateKey: Buffer | Uint8Array | string, message: BigNumber): Signature<bigint> {
```

Comment:
I believe the intent here is to change not only the types but the handling and checking of the inputs here.  A BigNumber is limited to bigint or a stringified bigint (not an arbitrary string, not a buffer).  The checkMessage function is the place where you should be checking that and converting to bigint appropriately.

Reply:
Also per @cedoor's comment on https://github.com/privacy-scaling-explorations/zk-kit/issues/230 there should be some documentation above about the fact that the message is a single BabyJub field element, represented as a bigint.  Best-practice for users with other types of input is probably to hash it (e.g. with a poseidon hash), or encode their inputs bytes directly if small enough (e.g. using bufferToBigInt()).